doc/conf.py

```diff
@@ -469,6 +469,15 @@ def add_js_css_files(app, pagename, templatename, context, doctree):
     "auto_examples/exercises/plot_cv_digits.py": (
         "auto_examples/model_selection/plot_nested_cross_validation_iris.py"
     ),
+    "auto_examples/cluster/plot_agglomerative_clustering.py": (
+        "auto_examples/cluster/plot_agglomerative_clustering_metrics.py"
+    ),
+    "auto_examples/cluster/plot_digits_agglomeration.py": (
+        "auto_examples/cluster/plot_feature_agglomeration_vs_univariate_selection.py"
+    ),
+    "auto_examples/cluster/plot_cluster_iris.py": (
+        "auto_examples/cluster/plot_kmeans_digits.py"
```

Comment:
```suggestion
    "auto_examples/cluster/plot_agglomerative_clustering": (
        "auto_examples/cluster/plot_agglomerative_clustering_metrics"
    ),
    "auto_examples/cluster/plot_digits_agglomeration": (
        "auto_examples/cluster/plot_feature_agglomeration_vs_univariate_selection"
    ),
    "auto_examples/cluster/plot_cluster_iris": (
        "auto_examples/cluster/plot_kmeans_digits"
```
I think `.py` should not be used, otherwise redirection will not work. You can check the redirection at [Check the rendered docs here! ](https://output.circle-artifacts.com/output/job/6b2e1307-7cd3-4cba-898b-656298635c82/artifacts/0/doc/_changed.html)

Reply:
Have already tested this, using `.py` doesn't cause an issue.

Reply:
It is. The redirection will not work properly when creating the HTML page.